keps/sig-scheduling/5055-dra-device-taints-and-tolerations/README.md

```diff
@@ -0,0 +1,787 @@
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- 3 examples of real-world usage
+- Allowing time for feedback
+- [Conformance tests]
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+### Upgrade / Downgrade Strategy
+
+Tainting gets disabled when downgrading to a release without support for it or
+when disabling the feature. The effect is as if the taints weren't set.
+
+### Version Skew Strategy
+
+During version skew where the apiserver supports the feature and the scheduler
+doesn't, taints can be set without encountering errors or
+warnings, but they won't have any effect.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+It is possible to disable the feature through the feature gate while leaving
+the API group enabled. This enables cleanup through the API.
+
+Re-enabling is supported because ResourceSlicePatches remain in etcd even if
+they are inaccessible and existing taints and tolerations are preserved during
+updates.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: DRADeviceTaints
+  - Components depending on the feature gate:
+    - kube-apiserver
+    - kube-scheduler
+    - kube-controller-manager
+- [X] Other
+  - Describe the mechanism: resource.k8s.io/v1alpha3 API group
+  - Will enabling / disabling the feature require downtime of the control
+    plane? Yes, in the apiserver.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? No.
+
+###### Does enabling the feature change any default behavior?
+
+No.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. The behavior of scheduling changes when it was in use.
+Running applications are not affected.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+It takes effect again for scheduling and may evict pods.
+
+###### Are there any tests for feature enablement/disablement?
+
+This will be covered through unit tests for the apiserver and scheduler.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+###### What specific metrics should inform a rollback?
+
+<!--
```

Comment:
Both here and in the other document, I'd suggest thinking through potential metrics for measuring how this affects the workloads.